Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector.py

```diff
@@ -0,0 +1,403 @@
+            ValueError: If unknown event type or missing cursor and start time.
+            DemistoException: If API responds with an HTTP error status code.
+
+        Returns:
+            dict: The response JSON from the event endpoint.
+        """
+        if not (feature := EVENT_TYPE_MAPPING.get(event_type)):
+            raise ValueError(f'Invalid or unsupported 1Password event type: {event_type}.')
+
+        body: dict[str, Any]
+        if pagination_cursor:
+            demisto.debug(f'Requesting events of type: {event_type} using pagination cursor: {pagination_cursor}')
+            body = {'cursor': pagination_cursor}
+
+        elif from_date:
+            formatted_from_date: str = from_date.strftime(DATE_FORMAT)
+            demisto.debug(f'Requesting events of type: {event_type} using from date: {formatted_from_date}')
+            body = {'limit': results_per_page, 'start_time': formatted_from_date}
+
+        else:
+            raise ValueError("Either a 'pagination_cursor' or a 'from_date' need to be specified.")
+
+        return self._http_request(method='POST', url_suffix=f'/{feature}', json_data=body, raise_on_status=True)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_limit_param(params: dict[str, str], event_type: str) -> int:
+    """Gets the limit parameter for a given event type. The limit represents the maximum number of events per fetch.
+
+    Args:
+        params (dict): The instance configuration parameters.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+
+    Returns:
+        int: The maximum number of events per fetch.
+    """
+    param_name = event_type.lower().replace(' ', '_') + '_limit'
+    return arg_to_number(params.get(param_name)) or DEFAULT_MAX_EVENTS_PER_FETCH
+
+
+def get_unauthorized_event_types(auth_introspection_response: dict[str, Any], event_types: list[str]) -> list[str]:
+    """Checks if the bearer token has no access to any of the configured event types.
+
+    Args:
+        auth_introspection_response (dict): Response JSON from Client.introspect call.
+        event_types (list): List of event types from the integration configuration params.
+
+    Returns:
+        list[str]: List of unauthorized event types that the token does not have access to.
+    """
+    authorized_features = auth_introspection_response['features']
+    return [
+        event_type for event_type in event_types
+        if EVENT_TYPE_MAPPING.get(event_type) not in authorized_features
+    ]
+
+
+def add_fields_to_event(event: dict[str, Any], event_type: str):
+    """Sets the '_time' and 'event_type' fields to an event.
+
+    Args:
+        event (dict): Event dictionary with the new fields.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+    """
+    event_time = arg_to_datetime(event['timestamp'], required=True)
+    event['_time'] = event_time.strftime(DATE_FORMAT)  # type: ignore[union-attr]
```

Comment:
Will verify and update the code if needed

Reply:
Verified with the SIEM team, both `_time` and `_TIME` are valid.

I prefer to use `_time` since that is what we use in all but one event collector (FireEye ETP)